contracts/Marketplace.sol

```diff
@@ -212,11 +232,17 @@ contract Marketplace is Collateral, Proofs {
   }
 
   function proofEnd(bytes32 slotId) public view returns (uint256) {
+    Slot memory slot = _slot(slotId);
     uint256 end = _end(slotId);
-    if (!_slotAcceptsProofs(slotId)) {
-      return end < block.timestamp ? end : block.timestamp - 1;
+    RequestContext storage context = _context(slot.requestId);
+    if (_slotAcceptsProofs(slotId)) {
+      return end;
+    } else {
+      // Calculate the earliest ending between a slot and a request.
+      // Request endings are set, for eg, when a request fails.
+      uint256 earliestEnd = Math.min(end, context.endsAt);
```

Comment:
There should be no difference between slot end and request end; they should always be the same.